lua/nvim-tree/help.lua

```diff
@@ -0,0 +1,197 @@
+local keymap = require "nvim-tree.keymap"
+
+local PAT_MOUSE = "^<.*Mouse"
+local PAT_CTRL = "^<C%-"
+local PAT_SPECIAL = "^<.+"
+
+local WIN_HL = table.concat({
+  "Normal:NvimTreeNormal",
+  "CursorLine:NvimTreeCursorLine",
+}, ",")
+
+local M = {
+  config = {},
+
+  -- one and only buf/win
+  bufnr = nil,
+  winnr = nil,
+}
+
+--- Shorten and normalise a vim command lhs
+--- @param lhs string
+--- @return string
+local function tidy_lhs(lhs)
+  -- nvim_buf_get_keymap replaces leading "<" with "<lt>" e.g. "<lt>CTRL-v>"
+  lhs = lhs:gsub("^<lt>", "<")
+
+  -- shorten ctrls
+  if lhs:lower():match "^<ctrl%-" then
+    lhs = lhs:lower():gsub("^<ctrl%-", "<C%-")
+  end
+
+  -- uppercase ctrls
+  if lhs:lower():match "^<c%-" then
+    lhs = lhs:upper()
+  end
+
+  -- space is not escaped
+  lhs = lhs:gsub(" ", "<Space>")
+
+  return lhs
+end
+
+--- Remove prefix 'nvim-tree: '
+--- Hardcoded to keep default_on_attach simple
+--- @param desc string
+--- @return string
+local function tidy_desc(desc)
+  return desc and desc:gsub("^nvim%-tree: ", "") or ""
+end
+
+--- sort vim command lhs roughly as per :help index
+--- @param a string
+--- @param b string
+local function sort_lhs(a, b)
+  -- mouse first
+  if a:match(PAT_MOUSE) and not b:match(PAT_MOUSE) then
+    return true
+  elseif not a:match(PAT_MOUSE) and b:match(PAT_MOUSE) then
+    return false
+  end
+
+  -- ctrl next
+  if a:match(PAT_CTRL) and not b:match(PAT_CTRL) then
+    return true
+  elseif not a:match(PAT_CTRL) and b:match(PAT_CTRL) then
+    return false
+  end
+
+  -- special next
+  if a:match(PAT_SPECIAL) and not b:match(PAT_SPECIAL) then
+    return true
+  elseif not a:match(PAT_SPECIAL) and b:match(PAT_SPECIAL) then
+    return false
+  end
+
+  -- remainder alpha
+  return a:gsub("[^a-zA-Z]", "") < b:gsub("[^a-zA-Z]", "")
+end
+
+--- Compute all lines for the buffer
+--- @return table strings of text
+--- @return table arrays of arguments 3-6 for nvim_buf_add_highlight()
+--- @return number maximum length of text
+local function compute()
+  local lines = { "nvim-tree mappings   exit: q" }
+  local hl = { { "NvimTreeRootFolder", 0, 0, 18 } }
+  local width = 0
+
+  -- formatted lhs and desc from active keymap
+  local mappings = vim.tbl_map(function(map)
+    return { lhs = tidy_lhs(map.lhs), desc = tidy_desc(map.desc) }
+  end, keymap.get_keymap())
+
+  -- sort roughly by lhs
+  table.sort(mappings, function(a, b)
+    return sort_lhs(a.lhs, b.lhs)
+  end)
+
+  -- longest lhs and description
+  local max_lhs = 0
+  local max_desc = 0
+  for _, l in pairs(mappings) do
+    max_lhs = math.max(#l.lhs, max_lhs)
+    max_desc = math.max(#l.desc, max_desc)
+  end
+
+  local fmt = string.format("%%-%ds %%-%ds", max_lhs, max_desc)
```

Comment:
```suggestion
  local fmt = string.format(" %%-%ds %%-%ds", max_lhs, max_desc)
```

Reply:
Adding padding after removal of border and moving window all the way to the edge.